### PR TITLE
Feat/cu 3ewevxy add tests to anythink backends 2

### PR DESCRIPTION
**Pull request by esapir**
# Description

Added tests for all backend routes.

All fixed a bunch of issues in rails and python. Some are bugs, some are alignments with node.

### CODE REVIEW

File: tests/e2e/anytinkClient.js [hunk 1]
```diff
@@ -8,41 +15,204 @@ class AnythinkClient {
...
   }
 
-  async createItem(item) {
-    try {
-      const itemRes = await this.client.post(`/api/items`, { item });
-      return itemRes.data?.item;
-    } catch (error) {
-      console.error("Create Item request failed", error?.response?.data);
-      throw error;
+  async getUserItems(seller, limit, offset, favorited, tag, callingUser) {
```
Comment (NivRichter) [inline tests/e2e/anytinkClient.js:149]: would be nice to be make it named params but probably doesn't matter much

File: yarn.lock [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (NivRichter) [inline yarn.lock:1]: Do we need this file?

**Review by NivRichter — APPROVED** [review]

File: .framework/rails/backend/app/controllers/items_controller.rb [hunk 3]
```diff
@@ -45,7 +45,7 @@ def feed
 
     @items_count = @items.count
 
-    @items = @items.order(created_at: :desc).offset(params[:offset] || 0).limit(params[:limit] || 20)
+    @items = @items.order(created_at: :asc).offset(params[:offset] || 0).limit(params[:limit] || 20)
```
[the remark this answers is not in the document]
Comment (esapir) [reply .framework/rails/backend/app/controllers/items_controller.rb:48]: All frameworks are now aligned.

File: tests/e2e/items.test.js [hunk 4]
```diff
@@ -0,0 +1,581 @@
...
+      const updateInfo = {
+        title: "New Title",
+        description: "New Description",
+        image: randomImageUrl(),
+      };
+
+      await createAndValidateUpdate(updateInfo);
+    });
+
+    const createAndValidateUpdate = async (updateInfo) => {
```
[the remark this answers is not in the document]
Comment (esapir) [reply tests/e2e/items.test.js]: :)

File: yarn.lock (hunk 2 above)
Comment (esapir) [reply yarn.lock]: removed

File: tests/e2e/anytinkClient.js (hunk 1 above)
Comment (esapir) [reply tests/e2e/anytinkClient.js:149]: Did not want just one or two functions to be with names params, and APIs are easier to use without it

**Pull request merged**